File: frontend/src/types/local-session.ts
export interface WorkspaceMatch {
  workspace_id: string;
  workspace_name: string;
  local_path: string;
}

export interface LocalSessionMeta {
  session_id: string;
  project_dir: string;
  cwd: string;
  git_branch: string | null;
  slug: string | null;
  version: string | null;
  first_timestamp: string | null;
  last_timestamp: string | null;
  file_size: number;
  message_count: number;
  already_imported: boolean;
  continuation_ids: string[];
  matched_workspace: WorkspaceMatch | null;
}

export interface ImportLocalSessionRequest {
  session_id: string;
  project_dir: string;
  workspace_id?: string | null;
}

export interface ImportLocalSessionResponse {
  dashboard_session_id: string;
  claude_session_id: string;
  messages_imported: number;
}

File: backend/app/schemas/local_session.py
"""로컬 세션 스캔/import Pydantic 스키마."""

from pydantic import BaseModel


class WorkspaceMatch(BaseModel):
    workspace_id: str
    workspace_name: str
    local_path: str


class LocalSessionMeta(BaseModel):
    session_id: str
    project_dir: str
    cwd: str
    git_branch: str | None = None
    slug: str | None = None
    version: str | None = None
    first_timestamp: str | None = None
    last_timestamp: str | None = None
    file_size: int = 0
    message_count: int = 0
    already_imported: bool = False
    continuation_ids: list[str] = []
    matched_workspace: WorkspaceMatch | None = None


class ImportLocalSessionRequest(BaseModel):
    session_id: str
    project_dir: str
    workspace_id: str | None = None


class ImportLocalSessionResponse(BaseModel):
    dashboard_session_id: str
    claude_session_id: str
    messages_imported: int
